### PR TITLE
feat(content): Add pocket trending stories for test-pilot add-on 

### PR DESCRIPTION
**Pull request by csadilek**
This PR brings in the Trending Stories by Pocket experiment. The experiment is disabled for now until we figure out the remaining open questions i.e. how should "Save to Pocket" and "Bookmark" work?

The PR includes:
- A new feed that fetches content recommendations from Pocket
- A new component that reuses the Spotlight widget to display pocket stories
- A minor enhancement to the Spotlight component to use all available space for the description if no image is available
- A new feed that produces system tick events, which can be used to periodically check for data expiry
- Minor enhancements to the NewTabPage and PreferencePane to bring in the PocketStories component and make it toggleable
- Test coverage for all of the above
- An enhancement to the build script to create a pocket.json file if none exists, so that API key and endpoint can be configured

Here's a screenshot showing the functionality if the experiment is enabled:
activity-stream-pocket

### CODE REVIEW

**Comment by csadilek** [comment]
@k88hudson hey kate, changes pretty much as discussed, can you take a look at this, please ;).

File: common/constants.js [hunk 1]
```diff
@@ -35,10 +38,14 @@ module.exports = {
   NEWTAB_PREFS_ENCODING: {
     "showSearch": 1 << 0,
     "showTopSites": 1 << 1,
-    "showHighlights": 1 << 2,
-    "showMoreTopSites": 1 << 3
+    "showPocketStories": 1 << 2,
```
Comment (ncloudioj) [inline common/constants.js]: Please append the new item to the end this encoding table other than inserting it in the middle. Otherwise, it'll break the existing telemetry.

i.e. 
```
NEWTAB_PREFS_ENCODING: {
     "showSearch": 1 << 0,
     "showTopSites": 1 << 1,
     "showHighlights": 1 << 2,
    "showMoreTopSites": 1 << 3,
    "showPocketStories": 1 << 5
}
```

File: docs/data_dictionary.md [hunk 2]
```diff
@@ -144,7 +144,8 @@ The Activity Stream addon sends various types of pings to the backend (HTTPS POS
 | --- | --- |
 | `showSearch` | 1 |
 | `showTopSites` | 2 |
-| `showHighlights` | 4 |
-| `showMoreTopSites` | 8 |
+| `showPocketStories` | 4 |
```
Comment (ncloudioj) [inline docs/data_dictionary.md]: ditto.

File: common/constants.js (hunk 1 above)
Comment (csadilek) [inline common/constants.js]: @ncloudioj thanks, done!

Comment (ncloudioj) [inline common/constants.js]: No worries! Great feature 👍 , BTW!

**Comment by csadilek** [comment]
Just updated to rebase on the latest style changes....

File: experiments.json [hunk 3]
```diff
@@ -238,5 +238,20 @@
       "threshold": 0.2,
       "description": "Edit Top Sites is enabled"
     }
+  },
+  "pocketStories": {
+    "name": "Trending stories provided by Pocket",
+    "active": false,
```
Comment (k88hudson) [inline experiments.json]: This should be `true` so it actually shows up in our list of experiments – if you want to ship it off for users, just set the `threshold` to `0`

File: content-src/components/PocketStories/PocketStories.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        page: this.props.page,
+        source: "RECOMMENDED",
+        action_position: index
+      };
+      this.props.dispatch(actions.NotifyEvent(payload));
+    };
+  },
+
+  renderStories() {
+    const stories = this.props.stories.slice(0, this.props.length);
```
Comment (k88hudson) [inline content-src/components/PocketStories/PocketStories.js]: can we hide the whole thing if `stores.length < 1`?

Comment (csadilek) [inline content-src/components/PocketStories/PocketStories.js]: @k88hudson thanks, fixed in render()

**Comment by csadilek** [comment]
@k88hudson Updated with requested changes, and additions to add a new feed to fetch pocket trending topics, plus refactoring to share pocket feed functionality and tests. Can you take a another look? Thanks again!

**Comment by csadilek** [comment]
Also, experiment is active threshold set to 0 for now.

**Comment by k88hudson** [comment]
Awesome, looks great!! 🔥 🔥 🔥

**Comment by k88hudson** [comment]
Sorry one more thing -- can you change the commit message to copy our style, e.g.

```
feat(content): Add pocket integration for test-pilot add-on
```

**Comment by csadilek** [comment]
@k88hudson thanks, commit msg udpated.

File: experiments.json [hunk 5]
```diff
@@ -238,5 +238,20 @@
...
+    "active": true,
+    "description": "Display recommendations of trending stories by Pocket",
+    "control": {
+      "value": true,
+      "description": "Pocket trending stories active"
+    },
+    "variant": {
+      "id": "exp-017-pocketstories",
+      "value": true,
+      "threshold": 0,
```
Comment (jaredkerim) [inline experiments.json]: Please change this threshold to a non zero value, we usually use 0.2 as a reasonable size given our TestPilot population size.  If the experiment is not ready to be deployed, then set the threshold to 0.2 and set active to false.

Thanks!

Comment (k88hudson) [inline experiments.json]: If `active` is false, the experiment can't be manually turned on

Comment (k88hudson) [inline experiments.json]: I'd personally be ok with turning on for 20%, but @csadilek was thinking maybe we should wait until they get the sign-off from pocket folks on monday

Comment (csadilek) [inline experiments.json]: @jaredkerim @k88hudson yes, my preference would be to merge this and send a separate PR next week to activate the experiment once we get everyone's OK. I am fine either way though.

Comment (jaredkerim) [inline experiments.json]: Okay then yes let's leave it as it is for now.  We'll need to find a better way to be able to separate being able to manually flag something on vs setting it as an active experiment, but we can deal with that later.

Comment (k88hudson) [inline experiments.json]: Yeah I totally agree – I'll file a ticket 👍
